chain-extensions/types/assets/Cargo.toml

```diff
@@ -0,0 +1,45 @@
+[package]
+name = "assets-chain-extension-types"
+version = "0.1.0"
+authors = ["Stake Technologies <devops@stake.co.jp>"]
+edition = "2021"
+license = "Apache-2.0"
+homepage = "https://astar.network"
+repository = "https://github.com/AstarNetwork/astar-frame"
+description = "Types definitions forassets chain-extension"
+
+[dependencies]
+ink = { version = "~4.0.0", default-features = false, optional = true }
+ink_metadata = { version = "~4.0.0", features = ["derive"], optional = true }
+
+scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive"] }
+scale-info = { version = "2.3.0", default-features = false, features = ["derive"] }
+
+frame-system = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37", default-features = false, optional = true }
+pallet-contracts = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37", default-features = false, optional = true }
+sp-runtime = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.37", default-features = false, optional = true }
+
+[features]
+default = ["std"]
+substrate = [
+	"pallet-contracts",
+	"sp-runtime",
+	"frame-system",
+]
+ink-no-std = [
+	"ink",
+]
+std = [
+	"scale-info/std",
+	"scale/std",
+]
+substrate-std = [
+	"std",
+	"sp-runtime/std",
+	"frame-system/std",
+	"pallet-contracts/std",
+]
+ink-std = [
+	"std",
+	"ink/std",
+]
```

Comment:
We had a discussion about this before but here's my new take on it.

1.
The `default` features should ensure code compiles without any issues. When doing the uplift to `polkadot-v0.9.37` I had to change `default` to `substrate-std` in order for normal `check` to pass.

2.
My experience from the uplift was that there were conflicting features (`cargo check --all-features` would fail).
It would be good to avoid this.
https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features

Reply:
If I put substrate dep in std it will not build on ink! side.
So only solution is to remove ink! dep from this crate and have a separate crate for ink!. Then we will lose coupling of Outcome enum (that was the reason why we put it in same crate)

Reply:
If it makes for cleaner code, perhaps it's fine.
I do believe we should follow good design principles and right now this seems a bit messy.

How about having enum generated via a simple macro?
That way we have controlled code duplication.